mne_bids/read.py

```diff
@@ -652,45 +653,66 @@ def get_matched_empty_room(bids_basename, bids_root):
+
+    kind = 'meg'  # We're only concerned about MEG data here
     bids_fname = _make_bids_fname(bids_basename=bids_basename,
                                   bids_root=bids_root, kind=kind)
     _, ext = _parse_ext(bids_fname)
+    if ext == '.fif':
+        extra_params = dict(allow_maxshield=True)
+    else:
+        extra_params = None
 
     raw = read_raw_bids(bids_basename=bids_basename, bids_root=bids_root,
-                        kind='meg')
+                        kind=kind, extra_params=extra_params)
     if raw.info['meas_date'] is None:
-        raise ValueError('Measurement date not available. Cannot get matching'
-                         ' empty room file')
+        raise ValueError('Measurement date not available. Cannot get matching '
+                         'empty-room file')
 
     ref_date = raw.info['meas_date']
     if not isinstance(ref_date, datetime):
         # for MNE < v0.20
         ref_date = datetime.fromtimestamp(raw.info['meas_date'][0])
-    search_path = make_bids_folders(bids_root=bids_root, subject='emptyroom',
-                                    session='**', make_dir=False)
-    search_path = op.join(search_path, '**', '**%s' % ext)
-    er_fnames = glob.glob(search_path)
-
-    best_er_fname = None
-    min_seconds = np.inf
-    for er_fname in er_fnames:
-        params = _parse_bids_filename(er_fname, verbose=False)
-        dt = datetime.strptime(params['ses'], '%Y%m%d')
-        dt = dt.replace(tzinfo=ref_date.tzinfo)
-        delta_t = dt - ref_date
-        if abs(delta_t.total_seconds()) < min_seconds:
-            min_seconds = abs(delta_t.total_seconds())
-            best_er_fname = er_fname
-
-    if best_er_fname is None:
-        er_basename = None
-    else:
-        from mne_bids import make_bids_basename  # Avoid circular import.
 
-        params = _parse_bids_filename(best_er_fname, verbose='warning')
+    emptyroom_dir = pathlib.Path(make_bids_folders(bids_root=bids_root,
+                                                   subject='emptyroom',
+                                                   make_dir=False))
+
+    if not emptyroom_dir.exists():
+        return None
+
+    # Find the empty-room recording sessions.
+    emptyroom_session_dirs = [x for x in emptyroom_dir.iterdir()
+                              if x.is_dir() and str(x.name).startswith('ses-')]
+    if not emptyroom_session_dirs:  # No session sub-directories found
+        emptyroom_session_dirs = [emptyroom_dir]
+
+    # Now try to discover all recordings inside the session directories.
+
+    allowed_extensions = list(reader.keys())
+    # `.pdf` is just a "virtual" extension for BTi data (which is stored iside
+    # a dedicated directory that doesn't have an extension)
+    del allowed_extensions[allowed_extensions.index('.pdf')]
+
+    candidate_paths = []
+    for session_dir in emptyroom_session_dirs:
+        dir_contents = (session_dir / kind).glob(f'sub-emptyroom_*_{kind}*')
```

Comment:
what does "session_dir / kind" do? Is that like doing `op.join` ?

Reply:
Yep. `pathlib.Path` overloads the `/` operator and turns it into a system-independent separator.

The same line without `Path` would read:

```python
        dir_contents = glob.glob(op.join(session_dir, kind, f'sub-emptyroom_*_{kind}*'))
```

Reply:
In 9fc4e93 I've now changed the path joining to use `os.path.join`, and globbing happens via `glob.glob`. I've left in the few other `Path`-based operations bc I think they really make the code better in terms of readability. (`Path.iterdir()`, `Path.is_dir()`, `Path.name`) Would that be okay with you?

Reply:
Oh, I just realized it's not so easy, because it then breaks a few other places… grrrr

Reply:
Fixed in 1e31c36 but it's ugly.